### PR TITLE
Make the `pre-commit` hook fix formatting 

### PR DESCRIPTION
**Pull request by kienstra**
## Changes
* The pre-commit hook is now like Gutenberg's. It fixes formatting when possible and stages the fix before the commit. If it can't fix it, it blocks the commit.
* Removes the [husky](https://www.npmjs.com/package/husky) pre-commit hook package. It was outdated, so we either had to [migrate](https://github.com/typicode/husky-4-to-8) or remove it.
* So to avoid `husky`, this adds  the pre-commit hook with a `git` command in `package.json` `prepare`
* I'll still have to iterate on this. But maybe we could use this in other repos if it helps.

## Testing instructions
1. Do a fresh clone of this repo
2. `git checkout remove/husky`
3. `composer i && nvm use && npm i`
4. Create a JS formatting error. Maybe remove a space inside parentheses in a JS file.
5. Stage that change
6. `git commit -m "Should be fixed"`
7. Expected: That formatting error is fixed before being committed.

### CODE REVIEW

**Comment by kienstra** [comment]
This isn't working.

File: package.json [hunk 1]
```diff
@@ -112,7 +111,7 @@
     "lint:php": "composer lint",
     "lint:php:fix": "composer lint-fix",
     "packages-update": "wp-scripts packages-update",
-    "precommit": "./bin/pre-commit.sh",
+    "prepare": "git config --local core.hooksPath .githooks",
```
Comment (kienstra) [inline package.json:114]: This won't always work, but I like it more than importing [husky](https://www.npmjs.com/package/husky) and updating its versions.

Also, we won't need a `.husky/` directory, only `.githooks/`.

I don't like directories named after vendors, but we still have some.

**Comment by kienstra** [comment]
There aren't this many commits for this work, those are because this PR targeted another branch, and I merged that branch in.

**Comment by kienstra** [comment]
Sorry, this actually isn't ready for review. It has some bugs.

File: phpunit.xml [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+	<testsuites>
+		<testsuite name="unit">
+			<directory suffix=".php">./tests/php/Unit/Helpers/</directory>
+			<directory prefix="Test" suffix=".php">./tests/php/Unit/</directory>
+		</testsuite>
+		<testsuite name="integration">
+			<directory suffix=".php">./tests/php/Integration/Helpers/</directory>
+			<directory prefix="Test" suffix=".php">./tests/php/Integration/</directory>
+		</testsuite>
+	</testsuites>
```
Comment (kienstra) [inline phpunit.xml:17]: Not related, I just wanted to indent this with tabs, like `phpcs.xml`

**Comment by kienstra** [comment]
This PR din't have that many commits

Those are because this was branched off another PR, and I merged that in.

File: .circleci/config.yml [hunk 3]
```diff
@@ -27,8 +27,8 @@ commands:
 
 jobs:
   lint:
-    executor:
-      name: php
+    docker:
+      - image: cimg/php:8.2-node
```
Comment (kienstra) [inline .circleci/config.yml:31]: To make linting pass

**Comment by kienstra** [comment]
Sorry, this is ready for review again.

**Comment by kienstra** [comment]
Sorry for yet another GCB PR!

**Review by mike-day — APPROVED** [review]
This is working well, @kienstra! It's a nice touch to do this without relying on husky.

---

Before making the commit, I had linting errors:

Screenshot 2023-08-01 at 10 29 14 AM

---

After making the commit, the auto-fixable errors were all fixed and the commit succeeded:

Screenshot 2023-08-01 at 10 29 36 AM

**Comment by kienstra** [comment]
Hi @mike-day,
Wow, great to see how it fixed those linting errors on your machine! 

Thanks for the screenshots and for testing this.

**Pull request merged**